### PR TITLE
Update version to 1.0.0

### PR DESCRIPTION
**Pull request by cparisineti**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by NathanJBennett — APPROVED** [review]

File: sdk/communication/azure-communication-jobrouter/CHANGELOG.md [hunk 1]
```diff
@@ -130,7 +130,7 @@
 #### BestWorkerMode
 - Added setters to `ScoringRule` and `ScoringRuleOptions`
 
-## 1.0.0-beta.2 (Unreleased)
+## 1.0.0-beta.2 (2023-07-27)
```
Comment (mssfang) [inline sdk/communication/azure-communication-jobrouter/CHANGELOG.md:133]: https://repo1.maven.org/maven2/com/azure/azure-communication-jobrouter/

1.0.0-beta.1 is the latest released version. 

We should remove ## 1.0.0-beta.2 sections. and update the release date for 1.0.0.

Comment (cparisineti) [reply sdk/communication/azure-communication-jobrouter/CHANGELOG.md]: Thank you. Updated it.

File: sdk/communication/azure-communication-jobrouter/CHANGELOG.md [hunk 2]
```diff
@@ -130,7 +130,7 @@
...
```
[the remark this answers is not in the document]
Comment (cparisineti) [reply sdk/communication/azure-communication-jobrouter/CHANGELOG.md]: Corrected it.

File: sdk/communication/azure-communication-jobrouter/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Release History
 
-## 1.0.0
+## 1.0.0 (Unreleased)
```
Comment (mssfang) [inline sdk/communication/azure-communication-jobrouter/CHANGELOG.md]: don't forget to update the date here

File: sdk/communication/azure-communication-jobrouter/customization/pom.xml [hunk 4]
```diff
@@ -5,7 +5,7 @@
   <parent>
     <artifactId>azure-communication-jobrouter</artifactId>
     <groupId>com.azure</groupId>
-    <version>1.0.0-beta.2</version>
+    <version>1.0.0</version>
```
Comment (mssfang) [inline sdk/communication/azure-communication-jobrouter/customization/pom.xml:8]: Missing

**Review by mssfang — APPROVED** [review]

**Pull request merged**